app/models/journeyDomain/locationOfGoods/LocationOfGoodsDomain.scala

```diff
@@ -50,24 +42,30 @@ object LocationOfGoodsDomain {
   implicit val userAnswersReader: UserAnswersReader[LocationOfGoodsDomain] =
     LocationTypePage.reader.flatMap {
       typeOfLocation =>
-        val identifierReads = InferredIdentificationPage.reader orElse IdentificationPage.reader
-        identifierReads.flatMap {
-          case CustomsOfficeIdentifier => LocationOfGoodsV.userAnswersReader(typeOfLocation)
-          case EoriNumber              => LocationOfGoodsX.userAnswersReader(typeOfLocation)
-          case AuthorisationNumber     => LocationOfGoodsY.userAnswersReader(typeOfLocation)
-          case UnlocodeIdentifier      => LocationOfGoodsU.userAnswersReader(typeOfLocation)
-          case CoordinatesIdentifier   => LocationOfGoodsW.userAnswersReader(typeOfLocation)
-          case AddressIdentifier       => LocationOfGoodsZ.userAnswersReader(typeOfLocation)
-          case PostalCode              => LocationOfGoodsT.userAnswersReader(typeOfLocation)
+        typeOfLocation match {
+          case LocationType("B", _) => LocationOfGoodsY.userAnswersReader()
+          case _ =>
+            val identifierReads: UserAnswersReader[LocationOfGoodsIdentification] = InferredIdentificationPage.reader orElse IdentificationPage.reader
+            identifierReads.map(_.qualifier).flatMap {
+              case CustomsOfficeIdentifier => LocationOfGoodsV.userAnswersReader(typeOfLocation)
+              case EoriNumber              => LocationOfGoodsX.userAnswersReader(typeOfLocation)
+              case AuthorisationNumber     => LocationOfGoodsY.userAnswersReader()
+              case UnlocodeIdentifier      => LocationOfGoodsU.userAnswersReader(typeOfLocation)
+              case CoordinatesIdentifier   => LocationOfGoodsW.userAnswersReader(typeOfLocation)
+              case AddressIdentifier       => LocationOfGoodsZ.userAnswersReader(typeOfLocation)
+              case PostalCode              => LocationOfGoodsT.userAnswersReader(typeOfLocation)
+              case x                       => throw new Exception(s"Unexpected Location of goods identifier value $x")
+            }
         }
+
     }
 
   case class LocationOfGoodsV(
     typeOfLocation: LocationType,
     customsOffice: CustomsOffice
   ) extends LocationOfGoodsDomain {
 
-    override val qualifierOfIdentification: LocationOfGoodsIdentification = CustomsOfficeIdentifier
+    override val qualifierOfIdentification: LocationOfGoodsIdentification = LocationOfGoodsIdentification("V", "CustomsOfficeIdentifier")
```

Comment:
Not keen on this val. I suspect `qualifierOfIdentification` can be removed from all of these and the base trait